knimin/templates/pm_plate_map.html

```diff
@@ -0,0 +1,358 @@
+
+  // Pre-process data
+  // The coordinate of a well is represented by a scalar "col_row" to facilitate searching
+  var well2id = {};
+  for (var i = 0; i < layout.length; i++) {
+    well2id[layout[i]['col'].toString() + "_" + layout[i]['row'].toString()] = i;
+  }
+
+  $(document).ready(function() {
+
+    // Render display items
+    $(".chosen-select").chosen();
+    $("#plated_on").datepicker();
+    $.ajax({
+      url: "/pm_plate_map/",
+      data: {action: "emails"},
+      success: function(emails) {
+        emails = JSON.parse(emails)
+        for (var i = 0; i < emails.length; i++) {
+          $("#plated_by").append($("<option>", {value: emails[i], text: emails[i]}));
+        }
+        $("#plated_by").trigger("chosen:updated");
+      }
+    });
+    $.ajax({
+      url: "/pm_plate_map/",
+      data: {action: "plate_types"},
+      success: function(plate_types) {
+        plate_types = JSON.parse(plate_types)
+        for (var i = 0; i < plate_types.length; i++) {
+          $("#plate_types").append($("<option>", {value: plate_types[i], text: plate_types[i]}));
+        }
+        $("#plate_types").val(type["name"]);
+        $("#plate_types").trigger("chosen:updated");
+      }
+    });
+
+    if (id > 0) {  // View or edit an existing plate
+      $("#plate_id").html("Plate # " + id.toString() + ": ");
+      $("#plate_name").val(info["name"]);
+      $("#plated_by").val(info["email"]);
+      $("#plated_on").datepicker("setDate", info["created_on"]);
+      $("#spanSave").hide();
+    } else {  // Create a new plate
+      $("#plated_by").val(currentuser);
+      $("#plated_on").datepicker("setDate", new Date());
+      $("#spanNav").hide();
+    }
+
+    // Generate plate map
+    var table = document.getElementById("plate_map");
+    table.innerHTML = "";
+    var ncol = type["cols"];
+    var nrow = type["rows"];
+    for (var i = 0; i <= nrow; i++) {
+      var row = table.insertRow();
+      var row_html = "";
+      for (var j = 0; j <= ncol; j++) {
+        if (i == 0) {
+          if (j == 0) {
+            row_html += "<td width=20></td>";
+          } else {
+            row_html += "<td class='head'>" + j.toString() + "</td>";
+          }
+        } else if (j == 0) {
+          row_html += "<td class='head'>" + String.fromCharCode(64 + i); + "</td>";
+        } else {
+          var well = j.toString() + "_" + i.toString();
+          var sample = {"sample_id": ""};
+          if (well2id.hasOwnProperty(well)) {
+            sample = layout[well2id[well]];
+          }
+          // The sample ID is stored in attributes: data-sample-original (original ID from the database) and data-sample-current (current ID after the last edit).
+          row_html += ("<td><input class='cell', type='text', data-sample-original='" + sample["sample_id"] + "', data-sample-current='" + sample["sample_id"] + "'></td>");
+        }
+      }
+      row.innerHTML = row_html;
+    }
+
+    // Hide leading zeros
```

Comment:
Did you check with the wet lab if this is actually useful? Since the data has the leading zeros I think it is better if they're shown. There are going to be cases where removing the leading 0 is not going to be acceptable and it is going to mislead the user.